### PR TITLE
Get candidate list 

### PR DESCRIPTION
**Pull request by ValeryiaMIRON**
- Get candidate list from Api 
https://trello.com/c/hq6LQw6Q/101-fe-get-candidate-list-by-internship-id
Снимок экрана 2021-11-08 в 12 51 32 PM

### CODE REVIEW

File: src/store/commands/types.js [hunk 1]
```diff
@@ -5,4 +5,10 @@ const LOGIN = {
 };
 const LOGOUT = { SUCCESS: 'LOGOUT_SUCCESS' };
 
-export { LOGIN, LOGOUT };
+const GETCANDIDATELIST = {
```
Comment (nsukhotskaya) [inline src/store/commands/types.js:8]:
```suggestion
const GET_CANDIDATE_LIST = {
```

Comment (ValeryiaMIRON) [reply src/store/commands/types.js]: done

File: src/store/commands/types.js [hunk 2]
```diff
@@ -5,4 +5,10 @@ const LOGIN = {
 };
 const LOGOUT = { SUCCESS: 'LOGOUT_SUCCESS' };
 
-export { LOGIN, LOGOUT };
+const GET_CANDIDATE_LIST = {
+  REQUEST: 'GETCANDIDATELIST_REQUEST',
```
Comment (nsukhotskaya) [inline src/store/commands/types.js:9]:
```suggestion
  REQUEST: 'GET_CANDIDATE_LIST_REQUEST',
```

File: src/store/actions/getCandidateList.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { GET_CANDIDATE_LIST } from '../commands/types';
+
+export const getCandidateListRequest = () => ({
+  type: GET_CANDIDATE_LIST.REQUEST,
+  payload: null,
+});
+
+export const getCandidateListSuccess = (data) => ({
+  type: GET_CANDIDATE_LIST.SUCCESS,
+  payload: data,
```
Comment (nsukhotskaya) [inline src/store/actions/getCandidateList.js:10]:
```suggestion
  payload: { someName: data},
```

File: src/store/actions/getCandidateList.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+});
+
+export const getCandidateListFailure = (error) => ({
+  type: GET_CANDIDATE_LIST.FAILURE,
+  payload: error,
```
Comment (nsukhotskaya) [inline src/store/actions/getCandidateList.js:15]:
```suggestion
  payload: {error},
```

File: src/store/reducers/candidateList.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+const initialState = {
+  candidates: [],
+  isLoading: false,
+};
+
+export default function candidateList(state = initialState, action) {
+  switch (action.type) {
+    case GET_CANDIDATE_LIST.REQUEST:
+      return { ...state, isLoading: true };
+    case GET_CANDIDATE_LIST.SUCCESS:
```
Comment (nsukhotskaya) [inline src/store/reducers/candidateList.js:12]:
```suggestion
    case GET_CANDIDATE_LIST.SUCCESS:
    const { someName } = action.payload;
```

File: src/store/reducers/candidateList.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+      return {
+        ...state,
+        candidates: action.payload,
```
Comment (nsukhotskaya) [inline src/store/reducers/candidateList.js:15]:
```suggestion
        candidates: someName,
```

File: src/store/commands/types.js (hunk 2 above)
Comment (ValeryiaMIRON) [reply src/store/commands/types.js:9]: done

File: src/store/actions/getCandidateList.js (hunk 3 above)
Comment (ValeryiaMIRON) [reply src/store/actions/getCandidateList.js:10]: done

File: src/store/actions/getCandidateList.js (hunk 4 above)
Comment (ValeryiaMIRON) [reply src/store/actions/getCandidateList.js:15]: done

File: src/store/reducers/candidateList.js (hunk 5 above)
Comment (ValeryiaMIRON) [reply src/store/reducers/candidateList.js:12]: done

File: src/store/reducers/candidateList.js (hunk 6 above)
Comment (ValeryiaMIRON) [reply src/store/reducers/candidateList.js:15]: done

File: src/store/commands/types.js [hunk 7]
```diff
@@ -5,4 +5,10 @@ const LOGIN = {
 };
 const LOGOUT = { SUCCESS: 'LOGOUT_SUCCESS' };
 
-export { LOGIN, LOGOUT };
+const GET_CANDIDATE_LIST = {
+  REQUEST: 'GET_CANDIDATE_LIST_REQUEST',
+  SUCCESS: 'GET_CANDIDATE_LIST_SUCCESS',
+  FAILURE: 'GET_CANDIDATE_LIST_FAILURE',
+};
+
```
Comment (nsukhotskaya) [inline src/store/commands/types.js:13]: Please delete types.js file and move GET_CANDIDATE_LIST to file in action folder

Comment (ValeryiaMIRON) [reply src/store/commands/types.js]: done

**Review by nsukhotskaya — APPROVED** [review]

**Review by beliavskaja — APPROVED** [review]

**Review by beliavskaja — APPROVED** [review]

**Review by nsukhotskaya — APPROVED** [review]

**Pull request merged**